cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/CredentialIdentityManager.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.credential;
+
+import io.cdap.cdap.common.AlreadyExistsException;
+import io.cdap.cdap.common.BadRequestException;
+import io.cdap.cdap.common.NotFoundException;
+import io.cdap.cdap.proto.credential.CredentialIdentity;
+import io.cdap.cdap.proto.id.CredentialIdentityId;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Manager for {@link io.cdap.cdap.proto.credential.CredentialIdentity} resources.
+ */
+public interface CredentialIdentityManager {
```

Comment:
is this an interface because we'll have the default implementation and also a remote implementation?

If we won't have a remote one we can just have the concrete class and skip the interface.

Reply:
I think there may need to be a remote interface for the namespace creation hook. I can remove the interface for the profile manager though.

Reply:
On a second thought, I'll go ahead and remove the interfaces and add them back when necessary